x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/metric/SecurityCacheMetrics.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.metric;
+
+import org.elasticsearch.common.cache.Cache;
+import org.elasticsearch.telemetry.metric.LongWithAttributes;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class SecurityCacheMetrics {
+
+    public static <K, V> List<AutoCloseable> registerAsyncCacheMetrics(MeterRegistry registry, Cache<K, V> cache, CacheType type) {
+        final List<AutoCloseable> metrics = new ArrayList<>();
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                type.metricsPrefix + ".hit.total",
+                "Total number of cache hits.",
+                "count",
+                () -> new LongWithAttributes(cache.stats().getHits())
+            )
+        );
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                type.metricsPrefix + ".miss.total",
+                "Total number of cache misses.",
+                "count",
+                () -> new LongWithAttributes(cache.stats().getMisses())
+            )
+        );
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                type.metricsPrefix + ".eviction.total",
+                "Total number of cache evictions.",
+                "count",
+                () -> new LongWithAttributes(cache.stats().getEvictions())
+            )
+        );
+        metrics.add(
+            registry.registerLongAsyncCounter(
```

Comment:
I think this should be a gauge, i.e. use `registerLongGauge`.

Reply:
whoops... Absolutely, this should be `gauge` instead of `counter`. Thanks for catching this. 🍺

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/109078/commits/a6778a2c926f8f059208322effd7169a459f8e7f